the-collab-lab/tcl-20-whats-near-me
# Functional Search Box

## Description
- Created a search field on the map and table views when the user enters free text into the input field and submits by hitting enter or pressing the search icon
- Added haversine function to find the distance from the center for google search results
- Transformed google search results to match Wikipedia results 
- Render all the locations (wikipedia and nearby search results) to pins to the map based on photos supplied
- Lists all locations (wikipedia and nearby search results) in order of distance from the center

## Related Issue
https://the-collab-lab.atlassian.net/browse/TCL20-15

## Acceptance Criteria
- [x] A search field should be present on both the map and table views (probably in a header bar at the top of the screen)
- [x] Users should be able to make free text searches 
- [x] The user will enter a place name or address, then click/tap a button to initiate the search

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓  | :sparkles: New feature     |
| ✓  | :hammer: Refactoring       |
|     | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before



### After

<img width="708" alt="Screen Shot 2021-02-27 at 7 30 12 PM" src="https://user-images.githubusercontent.com/59700665/109404356-57e74780-7933-11eb-91dd-e02f8a10809a.png">
<img width="718" alt="Screen Shot 2021-02-27 at 7 30 23 PM" src="https://user-images.githubusercontent.com/59700665/109404360-5fa6ec00-7933-11eb-9162-2bf7c44fc08f.png">
<img width="712" alt="Screen Shot 2021-02-27 at 7 30 37 PM" src="https://user-images.githubusercontent.com/59700665/109404361-603f8280-7933-11eb-8b94-df957537c224.png">
<img width="716" alt="Screen Shot 2021-02-27 at 7 30 45 PM" src="https://user-images.githubusercontent.com/59700665/109404362-603f8280-7933-11eb-8f4e-ba90f8ed1682.png">

## Testing Steps / QA Criteria
If you enter a name or place input that doesn't exist in the area or gibberish it will return that no results in this area in red. 
<img width="466" alt="Screen Shot 2021-02-27 at 7 52 05 PM" src="https://user-images.githubusercontent.com/59700665/109404575-751d1580-7935-11eb-85ff-82733f284732.png">
If you then enter a name or place that does exist, the error message should go away and the new results should be returned along with the Wikipedia results that were already there.
If you then navigate to the list, the search results should be displayed along with the Wikipedia results for the current area in order or distance (nearest to furthest)
If you click on an icon or list item for a Wikipedia page result, it will include a link to the Wikipedia page. If it's a Google places result, no link will appear. 
You should get the same search functionality from hitting enter as from clicking on the magnifying glass icon.
